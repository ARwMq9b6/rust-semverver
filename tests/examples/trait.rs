pub mod old {
    pub trait Abc {
        type A;
        fn test(&self) -> u8;
        fn test2(&self) -> u8;
        fn test4() -> u8 {
            0
        }
        fn test5() -> u8 {
            0
        }
        fn test7() -> u8;
    }
}

pub mod new {
    pub trait Abc {
        type A;
        fn test(&self) -> u8;
        fn test3(&self) -> u8;
        fn test4() -> u8 {
            0
        }
        fn test6() -> u8 {
            0
        }
        fn test7() -> u16;
    }
}
