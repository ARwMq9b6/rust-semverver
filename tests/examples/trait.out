version bump: 1.0.0 -> (Breaking) -> 2.0.0
warning: Breaking changes in `Abc`
  --> $REPO_PATH/tests/examples/trait.rs:17:5
   |
17 | /     pub trait Abc {
18 | |         type A;
19 | |         fn test(&self) -> u8;
20 | |         fn test3(&self) -> u8;
...  |
27 | |         fn test7() -> u16;
28 | |     }
   | |_____^
   |
warning: Breaking (TraitItemRemoved { defaulted: false })
  --> $REPO_PATH/tests/examples/trait.rs:5:9
   |
5  |         fn test2(&self) -> u8;
   |         ^^^^^^^^^^^^^^^^^^^^^^
warning: Breaking (TraitItemRemoved { defaulted: true })
  --> $REPO_PATH/tests/examples/trait.rs:9:9
   |
9  | /         fn test5() -> u8 {
10 | |             0
11 | |         }
   | |_________^
warning: Breaking (TraitItemAdded { defaulted: false })
  --> $REPO_PATH/tests/examples/trait.rs:20:9
   |
20 |         fn test3(&self) -> u8;
   |         ^^^^^^^^^^^^^^^^^^^^^^
note: TechnicallyBreaking (TraitItemAdded { defaulted: true })
  --> $REPO_PATH/tests/examples/trait.rs:24:9
   |
24 | /         fn test6() -> u8 {
25 | |             0
26 | |         }
   | |_________^

warning: Breaking changes in `test7`
  --> $REPO_PATH/tests/examples/trait.rs:27:9
   |
27 |         fn test7() -> u16;
   |         ^^^^^^^^^^^^^^^^^^
   |
   = warning: Breaking (TypeChanged { error: Sorts(ExpectedFound { expected: u8, found: u16 }) })

